tests/waku_core/topics/test_sharding.nim

```diff
@@ -0,0 +1,153 @@
+        nsPubsubTopic.value() == NsPubsubTopic.staticSharding(ClusterId, 3)
+    
+    test "Generate Gen0 Shard with topic.generation==0":
+      # When we get a shard from it
+      let nsPubsubTopic = getShard(contentTopicFull)
+      
+      # Then the generated shard is valid
+      check:
+        nsPubsubTopic.value() == NsPubsubTopic.staticSharding(ClusterId, 3)
+    
+    test "Generate Gen0 Shard with topic.generation==other":
+      # When we get a shard from it
+      let nsPubsubTopic = getShard(contentTopicInvalid)
+      
+      # Then the generated shard is valid
+      check:
+        nsPubsubTopic.error() == "Generation > 0 are not supported yet"
+    
+    test "Generate Gen0 Shard invalid topic":
+      # When we get a shard from it
+      let nsPubsubTopic = getShard("invalid")
+
+      # Then the generated shard is valid
+      check:
+        nsPubsubTopic.error() == "invalid format: topic must start with slash"
+
+  suite "parseSharding":
+    test "contentTopics is ContentTopic":
+      # When calling with contentTopic as string
+      let topicMap = parseSharding(some(pubsubTopic04), contentTopicShort)
+
+      # Then the topicMap is valid
+      check:
+        topicMap.value() == {pubsubTopic04: @[contentTopicShort]}
+
+    test "contentTopics is seq[ContentTopic]":
+      # When calling with contentTopic as string seq
+      let topicMap = parseSharding(some(pubsubTopic04), @[contentTopicShort, "/0/foo/1/bar/proto"])
+
+      # Then the topicMap is valid
+      check:
+        topicMap.value() == {pubsubTopic04: @[contentTopicShort, "/0/foo/1/bar/proto"]}
+
+    test "pubsubTopic is none":
+      # When calling with pubsubTopic as none
+      let topicMap = parseSharding(PubsubTopic.none(), contentTopicShort)
+
+      # Then the topicMap is valid
+      check:
+        topicMap.value() == {pubsubTopic13: @[contentTopicShort]}
+
+    test "content parse error":
+      # When calling with pubsubTopic as none with invalid content
+      let topicMap = parseSharding(PubsubTopic.none(), "invalid")
+
+      # Then the topicMap is valid
+      check:
+        topicMap.error() == "Cannot parse content topic: invalid format: topic must start with slash"
+
+    test "pubsubTopic parse error":
+      # When calling with pubsubTopic as none with invalid content
+      let topicMap = parseSharding(some("invalid"), contentTopicShort)
+
+      # Then the topicMap is valid
+      check:
+        topicMap.error() == "Cannot parse pubsub topic: invalid format: must start with /waku/2"
+
+    test "pubsubTopic getShard error":
+      # When calling with pubsubTopic as none with invalid content
+      let topicMap = parseSharding(PubsubTopic.none(), contentTopicInvalid)
+
+      # Then the topicMap is valid
+      check:
+        topicMap.error() == "Cannot autoshard content topic: Generation > 0 are not supported yet"
+
+    xtest "catchable error on add to topicMap":
+      # TODO: Trigger a CatchableError or mock
+      discard
```

Comment:
IMO either remove, add an issue or do it.

Reply:
I've implemented a simple mocking mechanism in PR depending on this one. If you guys don't mind, I'd love for this to get merged first so there's no need to reorder the commits, and I'll finish this test case as soon as it's available.